jmclient/jmclient/wallet_rpc.py

```diff
@@ -1240,6 +1240,24 @@ def getseed(self, request, walletname):
             seedphrase, _ = self.services["wallet"].get_mnemonic_words()
             return make_jmwalletd_response(request, seedphrase=seedphrase)
 
+        @app.route('/wallet/<string:walletname>/signmessage', methods=['POST'])
+        def signmessage(self, request, walletname):
+            self.check_cookie(request)
+            if not self.services["wallet"]:
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                raise InvalidRequestFormat()
+
+            request_data = self.get_POST_body(request, ["hd_path", "message"])
+            result = wallet_signmessage(self.services["wallet"],
+                request_data["hd_path"], request_data["message"])
+            if type(result) == str:
```

Comment:
If you set `out_str` to `False` in the call above, then you will always get the tuple returned.

The thing is, `wallet_signmessage` returns a str *either* if there is an error, *or* if out_str is False, so that it can output the full info string (it's for the command line, so that's why it's default). Bad design really, though not very important. The errors are only raised if the input hd path is invalid or the message is missing. For this RPC use case that doesn't apply, we input those programmatically so there's no human error. Errors just get caught by the exception raises.

So TLDR I don't think you need this if/else, but you do need to set `out_str` to `False`.

Reply:
Ohh, my mistake, somehow didn't notice that default is `True` not `False`. With `False` it does not always return tuple, it returns tuple with successful signing, string on error. So if/else is needed.

Yes, design here isn't the best, but, OTOH, it's not the most critical functionality, so maybe not worth spending too much time redesigning it.

Addressed in 59ee058934f322c04f27c3a3878128cf57608d11.